lib/fencing/st_client.c

```diff
@@ -2206,8 +2206,7 @@ stonith__later_succeeded(stonith_history_t *event, stonith_history_t *top_histor
          if ((prev_hp->state == st_done) &&
             pcmk__str_eq(event->target, prev_hp->target, pcmk__str_casei) &&
             pcmk__str_eq(event->action, prev_hp->action, pcmk__str_casei) &&
-            pcmk__str_eq(event->delegate, prev_hp->delegate, pcmk__str_casei) &&
-            ((event->completed < prev_hp->completed) || 
+            ((event->completed < prev_hp->completed) ||
```

Comment:
even cooler might be if we just memorize that we found a successful attempt via a different delegate and check if there is a success via the same delegate later on. like this we would still be pointed to cases where fencing via a certain delegate always fails. Like "a later attempt via a different delegate succeeded" or explicitly name the delegate that succeeded.
Unfortunately a simple boolean as return wouldn't do anymore. Maybe we could use the result-stuff for that purpose as well.

Reply:
stonith__later_succeeded() is only used by the "stonith-event" output message, so I can make it return NULL instead of false and the delegate instead of true, then the message can say "a later attempt from <delegate> succeeded". That assumes delegate will always be set, but I think that's true, and we can use something generic like "a cluster node" as a failsafe.